lcc_lokavaluto_app_connection/models/res_partner_backend.py

```diff
@@ -95,4 +97,16 @@ def get_wallet_balance(self):
             "success": False,
             "response": "No data - Please install financial backend Odoo add-on."
         }
-        return res
\ No newline at end of file
+        return res
+        
+    def get_wallet_commission_rule(self):
+        self.ensure_one()
+        rules = self.env["commission.rule"].search([("active", "=", True)])
+        for rule in rules:
+            # Get all the wallet matching the rule
+            wallets = self.search(safe_eval(rule.wallet_domain))
+            # Check if current wallet (self) is in the matching wallets
+            if wallets.filtered(lambda x: x.id == self.id):
+                # First rule matched is returned
+                return rule
```

Comment:
only the first rule will match... first in which order ? the default order as you didn't specify any `_order`. Is that what you want ? This computation will also be quite costly written like that. You should add the `('id', '=', self.id)` to the search at least and avoid doing the test in python.

Reply:
the idea is that as soon as one rule match, we consider that the reconversion is allowed, whatever the rule is.
I have modified the code to make it less costly.